test/integration/scheduler_perf/scheduler_test.go

```diff
@@ -253,3 +254,124 @@ func schedulePods(config *testConfig) int32 {
+	numGroups := nodeAffinity.numGroups
+	nodeAffinityKey := nodeAffinity.nodeAffinityKey
+	podList := make([]*v1.Pod, 0, numGroups)
+	for i := 0; i < numGroups; i++ {
+		pod = &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "sched-perf-node-affinity-pod-",
+			},
+			Spec: testutils.MakePodSpec(),
+		}
+		pod.Spec.Affinity = &v1.Affinity{
+			NodeAffinity: &v1.NodeAffinity{
+				RequiredDuringSchedulingIgnoredDuringExecution: &v1.NodeSelector{
+					NodeSelectorTerms: []v1.NodeSelectorTerm{
+						{
+							MatchExpressions: []v1.NodeSelectorRequirement{
+								{
+									Key:      nodeAffinityKey,
+									Operator: nodeAffinity.Operator,
+									Values:   []string{fmt.Sprintf("%v", i)},
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+		podList = append(podList, pod)
+	}
+	return podList
+}
+
+// Interface that every predicate or priority structure should implement.
+type UpdateNodePodConfig interface {
+	// mutateNode mutates the node strategies
+	mutateNode()
+	// mutatePod mutates the pod configs
+	mutatePod()
+}
+
+
+// TODO: As of now, returning configs hardcoded, need to read from yaml or some other file. A lot to validate as well.
+func readInPriorityConfiguration() *schedulerPerfConfig {
+	return &schedulerPerfConfig{
+		NodeAffinity: nodeAffinity{
+			Enabled:         true,
+			numGroups:       10,
+			nodeAffinityKey: "kubernetes.io/sched-perf-node-affinity",
+			Operator:        v1.NodeSelectorOpIn,
+		},
+
+		InterpodAffinity: interpodAffinity{
+			Enabled:     false,
+			Operator:    metav1.LabelSelectorOpIn,
+			affinityKey: "security",
+			Labels:      map[string]string{"security": "S1"},
+			TopologyKey: "region",
+		},
+	}
+}
+
+func (pc *schedulerPerfConfig) mutate(config *testConfig) {
+	nodeAffinity := pc.NodeAffinity
+	podCreatorConfig := testutils.NewTestPodCreatorConfig()
+	var nodeStrategies []testutils.CountToStrategy
+	var pod *v1.Pod
+	var podList []*v1.Pod
+	// It seems this should be the last one to run as this
+	if nodeAffinity.Enabled {
+		// Mutate Node
+		nodeStrategies = nodeAffinity.mutateNode(config.numNodes)
+		// Mutate Pod
+		podList = nodeAffinity.mutatePod(config.numPods, pod)
+		numGroups := nodeAffinity.numGroups
+		for _, pod := range podList {
+			podCreatorConfig.AddStrategy("sched-perf-node-affinity", config.numPods/numGroups,
+				testutils.NewCustomCreatePodStrategy(pod),
+			)
+		}
+		config.nodePreparer = framework.NewIntegrationTestNodePreparer(
+			config.schedulerSupportFunctions.GetClient(),
+			nodeStrategies, "scheduler-perf-")
+		config.podCreator = testutils.NewTestPodCreator(config.schedulerSupportFunctions.GetClient(), podCreatorConfig)
+	}
+	return
+}
+
+func TestSample(t *testing.T) {
+	// TODO: Read yaml file which has information on nodes, pods etc.
+	// TODO: Validate the yaml file and convert it into data structure that we have.
+	config := baseConfig()
+	config.numNodes = nodes
+	config.numPods = pods
+	// Fill in priority Configuration
```

Comment:
these comments will probably result in nitting upstream, best to just drop them.